lib/API/DX/Device.cpp

```diff
@@ -746,32 +779,43 @@ class DXDevice : public offloadtest::Device {
 } // namespace
 
 llvm::Error InitializeDXDevices() {
+#ifdef _WIN32
 #ifndef NDEBUG
-  CComPtr<ID3D12Debug1> Debug1;
+  ComPtr<ID3D12Debug1> Debug1;
 
   if (auto Err = HR::toError(D3D12GetDebugInterface(IID_PPV_ARGS(&Debug1)),
                              "failed to create D3D12 Debug Interface"))
     return Err;
 
   Debug1->EnableDebugLayer();
   Debug1->SetEnableGPUBasedValidation(true);
+#endif
 #endif
 
-  CComPtr<IDXGIFactory2> Factory;
-  if (auto Err = HR::toError(CreateDXGIFactory2(0, IID_PPV_ARGS(&Factory)),
-                             "Failed to create DXGI Factory")) {
+  ComPtr<IDXCoreAdapterFactory> Factory;
+  if (auto Err = HR::toError(DXCoreCreateAdapterFactory(Factory.GetAddressOf()),
+                             "Failed to create DXCore Adapter Factory")) {
     return Err;
   }
-  for (unsigned AdapterIndex = 0;;) {
-    CComPtr<IDXGIAdapter1> Adapter;
 
-    HRESULT HR = Factory->EnumAdapters1(AdapterIndex++, &Adapter);
-
-    if (DXGI_ERROR_NOT_FOUND == HR)
-      return llvm::Error::success();
+  ComPtr<IDXCoreAdapterList> AdapterList;
+  GUID attributes[]{DXCORE_ADAPTER_ATTRIBUTE_D3D12_CORE_COMPUTE,
```

Comment:
Here too: https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly